### ISSUE TITLE
LoadError: type Nothing has no field _conj

### ISSUE THREAD

**Issue by Yue-Zhengyuan** [Jutho/TensorKit.jl#158]
On Julia 1.11.1, I get the following error when building TensorKit: 

```
(@v1.11) pkg> precompile
Precompiling project...
  ✗ TensorKit → TensorKitChainRulesCoreExt
  0 dependencies successfully precompiled in 1 seconds. 466 already precompiled.

ERROR: The following 1 direct dependency failed to precompile:

TensorKitChainRulesCoreExt 

Failed to precompile TensorKitChainRulesCoreExt [74f4665c-2d76-51f1-bd11-a86ebbb150d0] to "/Users/zhengyuanyue/.julia/compiled/v1.11/TensorKitChainRulesCoreExt/jl_uJjNv9".
ERROR: LoadError: type Nothing has no field _conj
Stacktrace:
 [1] getproperty(x::Nothing, f::Symbol)
   @ Base ./Base.jl:49
 [2] top-level scope
   @ ~/.julia/packages/TensorKit/59D34/ext/TensorKitChainRulesCoreExt/TensorKitChainRulesCoreExt.jl:19
 [3] include
   @ ./Base.jl:557 [inlined]
 [4] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
   @ Base ./loading.jl:2790
 [5] top-level scope
   @ stdin:5
in expression starting at /Users/zhengyuanyue/.julia/packages/TensorKit/59D34/ext/TensorKitChainRulesCoreExt/TensorKitChainRulesCoreExt.jl:1
in expression starting at stdin:
```

Package versions:
```
[07d1fe3e] TensorKit v0.12.6
⌅ [6aa20fa7] TensorOperations v4.1.1
```

**Comment by lkdvos** [comment]
Aha, I see that my "extension hacking" has finally come to bite me in the backside...

It looks like julia 1.11 made some changes to the order of loading the package extensions. Because of that, the TensorKitChainRulesCoreExt module is not finding the TensorOperationsChainRulesCoreExt module, thus missing the required definitions. In principle this is not too hard to fix, it's a matter of porting over the definitions of `_conj` and `trivtuple`, but then we need to decide if it is worth it to backport this to a version 0.12.7, or if we wait for the new version of TensorKit. Any thoughts @Jutho ?

**Comment by Jutho** [comment]
I think this is due to this: https://github.com/JuliaLang/julia/pull/55589

**Comment by Jutho** [comment]
I was also just looking into this. I do think porting over the required definitions is probably the quickest fix, as we will be starting to get many bug reports as people upgrade to Julia 1.11.1.

**Comment by lkdvos** [comment]
Ah right, I forgot about that... We did not do this yet in the 0.12 version though, so it might be possible to hotfix that one and rethink the approach for the new tensorkit versions

**Comment by lkdvos** [comment]
This should now be resolved both in the v0.12 series as well as the v0.13 series, so I am closing this now.
